Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h

```diff
@@ -200,6 +200,11 @@ namespace AZ::DocumentPropertyEditor
             {
                 return AZ::Reflection::ReadGenericAttributeToDomValue(instance, attribute).value_or(AZ::Dom::Value());
             }
+            // Handle the attribute providing an invokable function instead of the value directly
```

Comment:
I'm slightly concerned how this would affect attributes that are intended to by invokable always, like changeNotify. In that case, calling LegacyAttributeToDomValue wouldn't give the DomValue for the callable attribute, but instead attempt to call it itself to retrieve a value that was not intended. Have you tried making this the final else case, like putting it below at line 214?
Something like:
if (!reader.Read<AttributeType>(value))
{
   if (attribute->CanDomInvoke(Dom::Value(Dom::Type::Array)))
    {
        return attribute->DomInvoke(instance, Dom::Value(Dom::Type::Array));
    }
   else
   {
    return AZ::Dom::Value();
    }

that way it would attempt to read the regular value first, and if it failed, it would try to invoke the attribute to get the result it was looking for.

Reply:
Good call, updated!